applications/asset_tracker_v2/src/cloud/cloud_codec/json_common.c

```diff
@@ -0,0 +1,757 @@
+	cJSON *cfg_obj = cJSON_CreateObject();
+
+	if (cfg_obj == NULL) {
+		err = -ENOMEM;
+		goto exit;
+	}
+
+	err = json_add_bool(cfg_obj, CONFIG_DEVICE_MODE,
+			    data->active_mode);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		goto exit;
+	}
+
+	err = json_add_number(cfg_obj, CONFIG_GPS_TIMEOUT,
+			      data->gps_timeout);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		goto exit;
+	}
+
+	err = json_add_number(cfg_obj, CONFIG_ACTIVE_TIMEOUT,
+			      data->active_wait_timeout);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		goto exit;
+	}
+
+	err = json_add_number(cfg_obj, CONFIG_MOVE_RES,
+			      data->movement_resolution);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		goto exit;
+	}
+
+	err = json_add_number(cfg_obj, CONFIG_MOVE_TIMEOUT,
+			      data->movement_timeout);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		goto exit;
+	}
+
+	err = json_add_number(cfg_obj, CONFIG_ACC_THRESHOLD,
+			      data->accelerometer_threshold);
+	if (err) {
+		LOG_ERR("Encoding error: %d returned at %s:%d", err, __FILE__, __LINE__);
+		goto exit;
+	}
+
+	json_add_obj(parent, object_label, cfg_obj);
+
+	return 0;
+
+exit:
+	cJSON_Delete(cfg_obj);
+	return err;
+}
+
+void json_common_config_get(cJSON *parent, struct cloud_data_cfg *data)
+{
+	cJSON *gps_timeout = cJSON_GetObjectItem(parent, CONFIG_GPS_TIMEOUT);
+	cJSON *active = cJSON_GetObjectItem(parent, CONFIG_DEVICE_MODE);
+	cJSON *active_wait = cJSON_GetObjectItem(parent, CONFIG_ACTIVE_TIMEOUT);
+	cJSON *move_res = cJSON_GetObjectItem(parent, CONFIG_MOVE_RES);
+	cJSON *move_timeout = cJSON_GetObjectItem(parent, CONFIG_MOVE_TIMEOUT);
+	cJSON *acc_thres = cJSON_GetObjectItem(parent, CONFIG_ACC_THRESHOLD);
+
+	if (gps_timeout != NULL) {
+		data->gps_timeout = gps_timeout->valueint;
+	}
+
+	if (active != NULL) {
+		data->active_mode = active->valueint;
+	}
+
+	if (active_wait != NULL) {
+		data->active_wait_timeout = active_wait->valueint;
+	}
+
+	if (move_res != NULL) {
+		data->movement_resolution = move_res->valueint;
+	}
+
+	if (move_timeout != NULL) {
+		data->movement_timeout = move_timeout->valueint;
+	}
+
+	if (acc_thres != NULL) {
+		data->accelerometer_threshold = acc_thres->valuedouble;
+	}
+}
+
+int json_common_batch_data_add(cJSON *parent, enum json_common_buffer_type type, void *buf,
+			       size_t buf_count, const char *object_label)
+{
+	int err = 0;
+	cJSON *array_obj = cJSON_CreateArray();
+
+	if (parent == NULL || array_obj == NULL) {
+		cJSON_Delete(array_obj);
+		return -ENOMEM;
+	}
+
+	if (object_label == NULL) {
+		LOG_WRN("Missing object label");
+		cJSON_Delete(array_obj);
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < buf_count; i++) {
+		switch (type) {
+		case JSON_COMMON_UI: {
+			struct cloud_data_ui *data =
+					(struct cloud_data_ui *)buf;
```

Comment:
Weird, I though that checkpatch requires an empty line after variable declaration. But as it doesn't complain I guess it's ok...

Reply:
Yeah, I noticed that too but If it makes the code clearer I can certainly add the extra lines.